### PR TITLE
Use WaitForInputIdle to ensure that the process is waiting for messages.

### LINKED ISSUE #2: Wait for target process to be ready before hooking
Call `Process.WaitForInputIdle()` before attempting to hook.

### PR DESCRIPTION
**Pull request by C1rdec**

### CODE REVIEW

**Comment by macote** [comment]
Is this a problem that you're experiencing in your project? I'm not sure I want to add this code in Winook since this kind of problem should be resolved beforehand.

**Comment by C1rdec** [comment]
> Is this a problem that you're experiencing in your project? I'm not sure I want to add this code in Winook since this kind of problem should be resolved beforehand.

There's no problem but everytime `Install` is called we need to make sure that the process is in the `IdleState` by calling `WaitForInputIdle`. I thought the code should be in the `Install` method since it make no sense to "Install" a hook on a process that it's not ready. 

It was just a suggestion :)

**Comment by macote** [comment]
It's a good suggestion. I read more about `WaitForInputIdle` and it does make sense to use it in this context. I saw a few things in your code that can be improved. Comments will follow.

**Comment by C1rdec** [comment]
You can modify the branch directly ;)
I gave you the permission.

File: src/Winook/HookBase.cs [hunk 1]
```diff
@@ -107,6 +107,17 @@ private void ReleaseAndDisposeMutex()
             }
         }
 
+        private void WaitForInputIdle()
+        {
+            try
```
Comment (macote) [inline src/Winook/HookBase.cs]: Don't use try/catch here. Let the exception bubble up.

File: src/Winook/HookBase.cs [hunk 2]
```diff
@@ -107,6 +107,17 @@ private void ReleaseAndDisposeMutex()
...
+            {
+                _targetProcess.WaitForInputIdle(2000);
```
Comment (macote) [inline src/Winook/HookBase.cs]: Use a constant defined at the top.

Comment (macote) [inline src/Winook/HookBase.cs]: You should either process the bool returned by `WaitForInputIdle(Int32)` or put a comment that indicates why you're ignoring it.

**Review by macote — APPROVED** [review]

**Pull request merged**